gtk4/src/dialog.rs

```diff
@@ -49,9 +53,30 @@ impl<O: IsA<Dialog>> DialogExtManual for O {
     fn get_response_for_widget<P: IsA<Widget>>(&self, widget: &P) -> ResponseType {
         unsafe {
             from_glib(ffi::gtk_dialog_get_response_for_widget(
-                self.as_ref().to_glib_none().0,
+                AsRef::<Dialog>::as_ref(self).to_glib_none().0,
                 widget.as_ref().to_glib_none().0,
             ))
         }
     }
+
+    fn run_future<'a>(&'a self) -> Pin<Box<dyn Future<Output = ResponseType> + 'a>> {
+        Box::pin(async move {
+            let (sender, receiver) = futures_channel::oneshot::channel();
+
+            let sender = std::cell::Cell::new(Some(sender));
+
+            let response_handler = self.connect_response(move |_, response_type| {
+                if let Some(m) = sender.replace(None) {
+                    let _result = m.send(response_type);
+                }
+            });
+
+            self.show();
```

Comment:
Do we want to keep this? It doesn't seem too unreasonable.

Reply:
I'd make it behave like `gtk_dialog_run()` in GTK3. Least amount of surprises :) I don't know what exactly it did back then